src/kOS/Suffixed/StageValues.cs

```diff
@@ -1,28 +1,54 @@
 ﻿using System;
+using System.Linq;
 using kOS.Safe.Encapsulation;
-using kOS.Utilities;
+using kOS.Safe.Encapsulation.Suffixes;
 
 namespace kOS.Suffixed
 {
     public class StageValues : Structure
     {
-        private readonly Vessel vessel;
+        private readonly SharedObjects shared;
 
-        public StageValues(Vessel vessel)
+        public StageValues(SharedObjects shared)
         {
-            this.vessel = vessel;
+            this.shared = shared;
+
+            InitializeSuffixes();
+        }
+
+        private void InitializeSuffixes()
+        {
+            AddSuffix("RESOURCES", new Suffix<SuffixedList<ActiveResourceValue>>(GetResourceManifest));
+        }
+
+        private SuffixedList<ActiveResourceValue> GetResourceManifest()
+        {
+            var resources = shared.Vessel.GetActiveResources();
+            var toReturn = new SuffixedList<ActiveResourceValue>();
+
+            foreach (var resource in resources)
+            {
+                toReturn.Add(new ActiveResourceValue(resource, shared));
+            }
+
+            return toReturn;
         }
 
         public override object GetSuffix(string suffixName)
         {
-            return GetResourceOfCurrentStage(suffixName);
+            var resourceAmount = GetResourceOfCurrentStage(suffixName);
+            if (resourceAmount.HasValue)
+            {
+                return resourceAmount.Value;
+            }
+
+            return base.GetSuffix(suffixName);
         }
 
-        private object GetResourceOfCurrentStage(string resourceName)
+        private double? GetResourceOfCurrentStage(string resourceName)
         {
-            var activeEngines = VesselUtils.GetListOfActivatedEngines(vessel);
-            var total = Utils.ProspectForResource(resourceName, activeEngines);
-            return Math.Round(total, 2);
+            var resource = shared.Vessel.GetActiveResources().FirstOrDefault(r => string.Equals(r.info.name, resourceName, StringComparison.OrdinalIgnoreCase));
+            return resource == null ? null : (double?) Math.Round(resource.amount, 2);
```

Comment:
Is the meaning of GetActiveResources just the ones in the current stage plus the ones that are stage-less?  (stage-less meaning things like ElectricCharge where all stages can always access the entire full ship's supply).